mitogen/core.py

```diff
@@ -3767,7 +3767,7 @@ def __init__(self, config):
         self.config = config
 
     def _on_broker_exit(self):
-        if not self.config['profiling']:
+        if not self.config['profiling'] and not hasattr(mitogen, "exit_status"):
             os.kill(os.getpid(), signal.SIGTERM)
```

Comment:
This part of the change is hacky and was meant to be temporary. I'd appreciate any suggestions on replacing it.

I think I will want to the replace the `mitogen.exit_status` variable with a less general `fakessh.exit_status` variable, and add a separate `self.config['term_on_exit']`option to control whether SIGTERM is sent when the broker shuts down.